src/linting/checks.jl

```diff
@@ -58,8 +61,9 @@ const LintCodeDescriptions = Dict{LintCodes,String}(IncorrectCallArgs => "Possib
     UnassignedKeywordArgument => "Keyword argument not assigned.",
     CannotDefineFuncAlreadyHasValue => "Cannot define function ; it already has a value.",
     DuplicateFuncArgName => "Function argument name not unique.",
-    IncludePathContainsNULL => "Cannot include file, path cotains NULL characters."
-    )
+    IncludePathContainsNULL => "Cannot include file, path cotains NULL characters.",
+    LoopOverLength => "Indexing into an array with indices obtained from `for i in 1:length(...)` and similar is discouraged. Use `eachindex` instead."
```

Comment:
Possible to have a better name since `LoopOverLength` shows in the UI? I don't  have a good suggestion though...

Reply:
Changed it to `ArrayIndexFromLength`.